Validate ShareableBitmap configuration

kkinnunen-apple opened the pull request:
#### 6f73b264931648ab04e902ddd3a1ec3335e82bd1
<pre>
Validate ShareableBitmap configuration
<a href="https://bugs.webkit.org/show_bug.cgi?id=298191">https://bugs.webkit.org/show_bug.cgi?id=298191</a>
<a href="https://rdar.apple.com/159603018">rdar://159603018</a>

Reviewed by NOBODY (OOPS!).

Validate the whole ShareableBitmapConfiguration. Only construct
valid instances. Remove unneeded validation after the construction.

Remove DestinationColorSpace ShareableBitmap::colorSpace().
ShareableBitmap might represent a bitmap that is not in a output color
space. In ShareableBitmapConfiguration, represent this with
platformColorSpace as WebKit doesn&apos;t have other type for arbitrary
colorspace.

For Cocoa:
Image pixel format is defined by bitsPerComponent, bitsPerPixel,
bitmapInfo and CGColorSpace. Transfer and validate these. The
CGBitmapInfo is only validated to be within the enumerations. CG
validates the combinations at the moment.

For Skia/Cairo:
Image pixel format is defined by &quot;is opaque&quot;.

* Source/WebCore/platform/graphics/Image.cpp:
(WebCore::Image::toShareableBitmap const):
* Source/WebCore/platform/graphics/ShareableBitmap.cpp:
(WebCore::ShareableBitmap::create):
(WebCore::ShareableBitmap::createFromImageDraw):
(WebCore::ShareableBitmapConfiguration::ShareableBitmapConfiguration): Deleted.
(WebCore::ShareableBitmapConfiguration::calculateSizeInBytes): Deleted.
* Source/WebCore/platform/graphics/ShareableBitmap.h:
(WebCore::ShareableBitmapConfiguration::headroom const):
(WebCore::ShareableBitmapConfiguration::bitsPerComponent const):
(WebCore::ShareableBitmapConfiguration::bitsPerPixel const):
(WebCore::ShareableBitmapConfiguration::size const):
(WebCore::ShareableBitmapConfiguration::colorSpace const):
(WebCore::ShareableBitmapConfiguration::bytesPerRow const):
(WebCore::ShareableBitmapConfiguration::sizeInBytes const):
(WebCore::ShareableBitmap::sizeInBytes const):
(WebCore::ShareableBitmap::create):
(WebCore::ShareableBitmapConfiguration::platformColorSpace const): Deleted.
(WebCore::ShareableBitmapConfiguration::isOpaque const): Deleted.
(WebCore::ShareableBitmapConfiguration::bytesPerPixel const): Deleted.
(WebCore::ShareableBitmapConfiguration::imageInfo const): Deleted.
(WebCore::ShareableBitmap::colorSpace const): Deleted.
* Source/WebCore/platform/graphics/cairo/ShareableBitmapCairo.cpp:
(WebCore::ShareableBitmapConfiguration::validateColorSpace): Deleted.
(WebCore::ShareableBitmapConfiguration::calculateBitsPerComponent): Deleted.
(WebCore::ShareableBitmapConfiguration::calculateBytesPerPixel): Deleted.
* Source/WebCore/platform/graphics/cg/ImageUtilitiesCG.cpp:
(WebCore::decodeImageWithSize):
* Source/WebCore/platform/graphics/cg/ShareableBitmapCG.mm:
(WebCore::resolveShareableBitmapConfiguration):
(WebCore::ShareableBitmapConfiguration::create):
(WebCore::ShareableBitmapConfiguration::ShareableBitmapConfiguration):
(WebCore::ShareableBitmap::create):
(WebCore::ShareableBitmapConfiguration::calculateBytesPerRow):
(WebCore::ShareableBitmap::createFromImagePixels):
(WebCore::ShareableBitmap::createGraphicsContext):
(WebCore::ShareableBitmap::makeCGImage):
(WebCore::ShareableBitmapConfiguration::validateColorSpace): Deleted.
(WebCore::ShareableBitmapConfiguration::calculateBitsPerComponent): Deleted.
(WebCore::ShareableBitmapConfiguration::calculateBytesPerPixel): Deleted.
(WebCore::ShareableBitmapConfiguration::calculateBitmapInfo): Deleted.
(WebCore::ShareableBitmap::createCGImage const): Deleted.
* Source/WebCore/platform/graphics/skia/ShareableBitmapSkia.cpp:
(WebCore::ShareableBitmapConfiguration::validateColorSpace): Deleted.
(WebCore::ShareableBitmapConfiguration::calculateBitsPerComponent): Deleted.
(WebCore::ShareableBitmapConfiguration::calculateBytesPerPixel): Deleted.
* Source/WebKit/GPUProcess/graphics/RemoteImageBuffer.cpp:
(WebKit::RemoteImageBuffer::getShareableBitmap):
(WebKit::RemoteImageBuffer::filteredNativeImage):
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.cpp:
(WebKit::RemoteRenderingBackend::cacheNativeImage):
* Source/WebKit/GPUProcess/graphics/wc/WCScene.cpp:
(WebKit::WCScene::update):
* Source/WebKit/GPUProcess/media/RemoteImageDecoderAVFProxy.cpp:
(WebKit::RemoteImageDecoderAVFProxy::createFrameImageAtIndex):
* Source/WebKit/GPUProcess/media/RemoteMediaPlayerManagerProxy.cpp:
(WebKit::RemoteMediaPlayerManagerProxy::bitmapImageForCurrentTime):
* Source/WebKit/Shared/ContextMenuContextData.cpp:
(WebKit::ContextMenuContextData::setImage):
(WebKit::ContextMenuContextData::setPotentialQRCodeNodeSnapshotImage):
(WebKit::ContextMenuContextData::setPotentialQRCodeViewportSnapshotImage):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.mm:
(WebKit::RemoteLayerBackingStoreProperties::layerContentsBufferFromBackendHandle):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::convertPlatformImageToBitmap):
* Source/WebKit/WebProcess/GPU/graphics/ImageBufferShareableBitmapBackend.cpp:
(WebKit::ImageBufferShareableBitmapBackend::calculateSafeBackendSize):
(WebKit::ImageBufferShareableBitmapBackend::create):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::snapshot):
* Source/WebKit/WebProcess/WebCoreSupport/ShareableBitmapUtilities.cpp:
(WebKit::createShareableBitmap):
* Source/WebKit/WebProcess/WebCoreSupport/gtk/WebDragClientGtk.cpp:
(WebKit::convertCairoSurfaceToShareableBitmap):
(WebKit::convertSkiaImageToShareableBitmap):
* Source/WebKit/WebProcess/WebCoreSupport/mac/WebDragClientMac.mm:
(WebKit::convertDragImageToBitmap):
* Source/WebKit/WebProcess/WebCoreSupport/win/WebPopupMenuWin.cpp:
(WebKit::WebPopupMenu::setUpPlatformData):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/AcceleratedSurface.cpp:
(WebKit::AcceleratedSurface::RenderTargetSHMImage::create):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.cpp:
(WebKit::DrawingAreaCoordinatedGraphics::display):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::loadAndDecodeImage):
(WebKit::WebPage::takeSnapshotForTargetedElement):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::drawToImage):
* Source/WebKit/WebProcess/cocoa/TextTrackRepresentationCocoa.mm:
(WebKit::WebTextTrackRepresentationCocoa::update):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6f73b264931648ab04e902ddd3a1ec3335e82bd1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/119593 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/39287 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/29939 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/125870 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/71667 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/00e1dae2-7660-4cb7-ab8c-77d90a098b4a) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/121469 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/39983 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/47865 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/90837 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/60139 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/2322fe19-81f3-4cd1-ab18-5640cce2ea10) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/122544 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/31916 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/107239 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/71346 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/3b50f518-6c75-4af0-ad2d-ae5acfb7e369) 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/30951 "Found 5 new test failures: imported/w3c/web-platform-tests/html/canvas/element/manual/wide-gamut-canvas/canvas-display-p3-drawImage-ImageBitmap-Blob.html imported/w3c/web-platform-tests/html/canvas/element/manual/wide-gamut-canvas/canvas-display-p3-drawImage-ImageBitmap-cloned.html imported/w3c/web-platform-tests/html/canvas/element/manual/wide-gamut-canvas/canvas-display-p3-drawImage-ImageBitmap-image.html imported/w3c/web-platform-tests/html/canvas/element/manual/wide-gamut-canvas/canvas-display-p3-drawImage.https.html imported/w3c/web-platform-tests/html/canvas/element/manual/wide-gamut-canvas/canvas-display-p3-pattern-image.html (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/25343 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/69520 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/111720 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/101382 "Passed tests") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25536 "Found 5 new test failures: imported/w3c/web-platform-tests/html/canvas/element/manual/wide-gamut-canvas/canvas-display-p3-drawImage-ImageBitmap-Blob.html imported/w3c/web-platform-tests/html/canvas/element/manual/wide-gamut-canvas/canvas-display-p3-drawImage-ImageBitmap-cloned.html imported/w3c/web-platform-tests/html/canvas/element/manual/wide-gamut-canvas/canvas-display-p3-drawImage-ImageBitmap-image.html imported/w3c/web-platform-tests/html/canvas/element/manual/wide-gamut-canvas/canvas-display-p3-drawImage.https.html imported/w3c/web-platform-tests/html/canvas/element/manual/wide-gamut-canvas/canvas-display-p3-pattern-image.html (failure)") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/128841 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/118111 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/46515 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/35234 "Found 60 new test failures: accessibility/aria-list-and-listitem.html accessibility/element-line-rects-and-text.html accessibility/empty-image-with-title.html accessibility/image-load-on-delay.html accessibility/mac/area-with-aria-label.html accessibility/mac/aria-liveregion-on-image.html accessibility/mac/aria-liveregions-changedalt.html accessibility/mac/aria-liveregions-notifications.html accessibility/mac/aria-liveregions-removedelement.html accessibility/mac/attributed-string-with-image.html ... (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/99434 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/46880 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103432 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/99276 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/44703 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/22723 "Found 10 new test failures: imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-left-of-viewport-offscreen-old.html imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-left-of-viewport-partially-onscreen-old.html imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-right-and-left-of-viewport-partially-onscreen-old.html imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-right-of-viewport-offscreen-old.html imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-right-of-viewport-partially-onscreen-old.html imported/w3c/web-platform-tests/html/canvas/element/manual/wide-gamut-canvas/canvas-display-p3-drawImage-ImageBitmap-Blob.html imported/w3c/web-platform-tests/html/canvas/element/manual/wide-gamut-canvas/canvas-display-p3-drawImage-ImageBitmap-cloned.html imported/w3c/web-platform-tests/html/canvas/element/manual/wide-gamut-canvas/canvas-display-p3-drawImage-ImageBitmap-image.html imported/w3c/web-platform-tests/html/canvas/element/manual/wide-gamut-canvas/canvas-display-p3-drawImage.https.html imported/w3c/web-platform-tests/html/canvas/element/manual/wide-gamut-canvas/canvas-display-p3-pattern-image.html (failure)") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/43079 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/46377 "Built successfully") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/52083 "Found 1 new failure in Shared/RemoteLayerTree/RemoteLayerBackingStore.mm") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/146809 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/45843 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37735 "Found 1 new JSC binary failure: testapi, Found 18666 new JSC stress test failures: ChakraCore.yaml/ChakraCore/test/Array/array_slice.js.default, ChakraCore.yaml/ChakraCore/test/Array/array_sort2.js.default, ChakraCore.yaml/ChakraCore/test/Array/reverse1.js.default, ChakraCore.yaml/ChakraCore/test/Array/toLocaleString.js.default, ChakraCore.yaml/ChakraCore/test/Bugs/blue_1086262.js.default, ChakraCore.yaml/ChakraCore/test/ControlFlow/forInPrimitive.js.default, ChakraCore.yaml/ChakraCore/test/Date/date_cache_bug.js.default, ChakraCore.yaml/ChakraCore/test/Error/CallNonFunction.js.default, ChakraCore.yaml/ChakraCore/test/Error/stack2.js.default, ChakraCore.yaml/ChakraCore/test/Function/FuncBody.bug227901.js.default ... (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/49192 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/47529 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d5e3d809ad1fbecffc964746a6fef96ce9878240. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50145

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/6d59362e22761f3fe72e21c9c21a45fec0d32cc3. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50145

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/f30fdede0c9425be3a5a0315adab93e40b5e8eba. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50145

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/eaca27e695d074daca949b859f21e81e5466c60d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50145

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/04c4bc7e559f61152ee34b791b7dce0161304ca9. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50145

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4533e423a0694f857f9b9a8d71cdcf153c183b36. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50145

kkinnunen-apple committed:
Validate ShareableBitmap configuration
https://bugs.webkit.org/show_bug.cgi?id=298191
rdar://159603018

Reviewed by NOBODY (OOPS!).

Validate the whole ShareableBitmapConfiguration. Only construct
valid instances. Remove unneeded validation after the construction.

Remove DestinationColorSpace ShareableBitmap::colorSpace().
ShareableBitmap might represent a bitmap that is not in a output color
space. In ShareableBitmapConfiguration, represent this with
platformColorSpace as WebKit doesn't have other type for arbitrary
colorspace.

For Cocoa:
Image pixel format is defined by bitsPerComponent, bitsPerPixel,
bitmapInfo and CGColorSpace. Transfer and validate these. The
CGBitmapInfo is only validated to be within the enumerations. CG
validates the combinations at the moment.

For Skia/Cairo:
Image pixel format is defined by "is opaque".

* Source/WebCore/platform/graphics/Image.cpp:
(WebCore::Image::toShareableBitmap const):
* Source/WebCore/platform/graphics/ShareableBitmap.cpp:
(WebCore::ShareableBitmap::create):
(WebCore::ShareableBitmap::createFromImageDraw):
(WebCore::ShareableBitmapConfiguration::ShareableBitmapConfiguration): Deleted.
(WebCore::ShareableBitmapConfiguration::calculateSizeInBytes): Deleted.
* Source/WebCore/platform/graphics/ShareableBitmap.h:
(WebCore::ShareableBitmapConfiguration::headroom const):
(WebCore::ShareableBitmapConfiguration::bitsPerComponent const):
(WebCore::ShareableBitmapConfiguration::bitsPerPixel const):
(WebCore::ShareableBitmapConfiguration::size const):
(WebCore::ShareableBitmapConfiguration::colorSpace const):
(WebCore::ShareableBitmapConfiguration::bytesPerRow const):
(WebCore::ShareableBitmapConfiguration::sizeInBytes const):
(WebCore::ShareableBitmap::sizeInBytes const):
(WebCore::ShareableBitmap::create):
(WebCore::ShareableBitmapConfiguration::platformColorSpace const): Deleted.
(WebCore::ShareableBitmapConfiguration::isOpaque const): Deleted.
(WebCore::ShareableBitmapConfiguration::bytesPerPixel const): Deleted.
(WebCore::ShareableBitmapConfiguration::imageInfo const): Deleted.
(WebCore::ShareableBitmap::colorSpace const): Deleted.
* Source/WebCore/platform/graphics/cairo/ShareableBitmapCairo.cpp:
(WebCore::ShareableBitmapConfiguration::validateColorSpace): Deleted.
(WebCore::ShareableBitmapConfiguration::calculateBitsPerComponent): Deleted.
(WebCore::ShareableBitmapConfiguration::calculateBytesPerPixel): Deleted.
* Source/WebCore/platform/graphics/cg/ImageUtilitiesCG.cpp:
(WebCore::decodeImageWithSize):
* Source/WebCore/platform/graphics/cg/ShareableBitmapCG.mm:
(WebCore::resolveShareableBitmapConfiguration):
(WebCore::ShareableBitmapConfiguration::create):
(WebCore::ShareableBitmapConfiguration::ShareableBitmapConfiguration):
(WebCore::ShareableBitmap::create):
(WebCore::ShareableBitmapConfiguration::calculateBytesPerRow):
(WebCore::ShareableBitmap::createFromImagePixels):
(WebCore::ShareableBitmap::createGraphicsContext):
(WebCore::ShareableBitmap::makeCGImage):
(WebCore::ShareableBitmapConfiguration::validateColorSpace): Deleted.
(WebCore::ShareableBitmapConfiguration::calculateBitsPerComponent): Deleted.
(WebCore::ShareableBitmapConfiguration::calculateBytesPerPixel): Deleted.
(WebCore::ShareableBitmapConfiguration::calculateBitmapInfo): Deleted.
(WebCore::ShareableBitmap::createCGImage const): Deleted.
* Source/WebCore/platform/graphics/skia/ShareableBitmapSkia.cpp:
(WebCore::ShareableBitmapConfiguration::validateColorSpace): Deleted.
(WebCore::ShareableBitmapConfiguration::calculateBitsPerComponent): Deleted.
(WebCore::ShareableBitmapConfiguration::calculateBytesPerPixel): Deleted.
* Source/WebKit/GPUProcess/graphics/RemoteImageBuffer.cpp:
(WebKit::RemoteImageBuffer::getShareableBitmap):
(WebKit::RemoteImageBuffer::filteredNativeImage):
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.cpp:
(WebKit::RemoteRenderingBackend::cacheNativeImage):
* Source/WebKit/GPUProcess/graphics/wc/WCScene.cpp:
(WebKit::WCScene::update):
* Source/WebKit/GPUProcess/media/RemoteImageDecoderAVFProxy.cpp:
(WebKit::RemoteImageDecoderAVFProxy::createFrameImageAtIndex):
* Source/WebKit/GPUProcess/media/RemoteMediaPlayerManagerProxy.cpp:
(WebKit::RemoteMediaPlayerManagerProxy::bitmapImageForCurrentTime):
* Source/WebKit/Shared/ContextMenuContextData.cpp:
(WebKit::ContextMenuContextData::setImage):
(WebKit::ContextMenuContextData::setPotentialQRCodeNodeSnapshotImage):
(WebKit::ContextMenuContextData::setPotentialQRCodeViewportSnapshotImage):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.mm:
(WebKit::RemoteLayerBackingStoreProperties::layerContentsBufferFromBackendHandle):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::convertPlatformImageToBitmap):
* Source/WebKit/WebProcess/GPU/graphics/ImageBufferShareableBitmapBackend.cpp:
(WebKit::ImageBufferShareableBitmapBackend::calculateSafeBackendSize):
(WebKit::ImageBufferShareableBitmapBackend::create):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::snapshot):
* Source/WebKit/WebProcess/WebCoreSupport/ShareableBitmapUtilities.cpp:
(WebKit::createShareableBitmap):
* Source/WebKit/WebProcess/WebCoreSupport/gtk/WebDragClientGtk.cpp:
(WebKit::convertCairoSurfaceToShareableBitmap):
(WebKit::convertSkiaImageToShareableBitmap):
* Source/WebKit/WebProcess/WebCoreSupport/mac/WebDragClientMac.mm:
(WebKit::convertDragImageToBitmap):
* Source/WebKit/WebProcess/WebCoreSupport/win/WebPopupMenuWin.cpp:
(WebKit::WebPopupMenu::setUpPlatformData):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/AcceleratedSurface.cpp:
(WebKit::AcceleratedSurface::RenderTargetSHMImage::create):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.cpp:
(WebKit::DrawingAreaCoordinatedGraphics::display):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::loadAndDecodeImage):
(WebKit::WebPage::takeSnapshotForTargetedElement):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::drawToImage):
* Source/WebKit/WebProcess/cocoa/TextTrackRepresentationCocoa.mm:
(WebKit::WebTextTrackRepresentationCocoa::update):
